core-test/src/main/java/org/openstack4j/api/artifact/ToscaTemplatesArtifactTests.java

```diff
@@ -0,0 +1,141 @@
+package org.openstack4j.api.artifact;
+
+import org.openstack4j.api.AbstractTest;
+import org.openstack4j.api.Builders;
+import org.openstack4j.model.artifact.ArtifactUpdate;
+import org.openstack4j.model.artifact.ToscaTemplatesArtifact;
+import org.openstack4j.model.artifact.ToscaTemplatesArtifacts;
+import org.openstack4j.model.artifact.builder.ArtifactUpdateBuilder;
+import org.openstack4j.model.artifact.builder.ToscaTemplatesArtifactBuilder;
+import org.openstack4j.model.common.ActionResponse;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.testng.Assert.assertEquals;
+
+/**
+ * Tests the Artifact -> Tosca Template Artifact API against the mock webserver and spec based
+ * json responses
+ *
+ * @author Pavan Vadavi
+ */
+@Test(suiteName = "ToscaTemplatesArtifact")
+public class ToscaTemplatesArtifactTests extends AbstractTest {
+
+    private static final String JSON_GET_ARTIFACT = "/artifact/tosca_templates_get_artifact.json";
+    private static final String JSON_CREATE_ARTIFACT = "/artifact/tosca_templates_create_artifact.json";
+    private static final String JSON_UPDATE_ARTIFACT = "/artifact/tosca_templates_update_artifact.json";
+    private static final String JSON_UPLOAD_ARTIFACT = "/artifact/tosca_templates_upload_artifact.json";
+    private static final String JSON_ACTIVATE_ARTIFACT = "/artifact/tosca_templates_activate_artifact.json";
+    private static final String JSON_DEACTIVATE_ARTIFACT = "/artifact/tosca_templates_deactivate_artifact.json";
+    private static final String JSON_PUBLISH_ARTIFACT = "/artifact/tosca_templates_publish_artifact.json";
+    private static final String ARTIFACT_ID = "b8f88696-80e7-4f89-abbc-1975991ba879";
+    private static final String JSON_ARTIFACTS = "/artifact/tosca_templates_artifacts.json";
+    private static final String ZIP_FILE_PATH = ".s /artifact/test.zip";
+
+    @Override
+    protected Service service() {
+        return Service.ARTIFACT;
+    }
+
+    @Test
+    public void getArtifact() throws Exception {
+        respondWith(JSON_GET_ARTIFACT);
+        ToscaTemplatesArtifact toscaTemplatesArtifact = osv3().artifact().toscaTemplatesArtifact().get(ARTIFACT_ID);
+        server.takeRequest();
```

Comment:
Minor: Shouldn't this work without `server.takeRequest();`?
(Also in the other tests)